src/api/FactoryAPI.ts

```diff
@@ -138,6 +144,78 @@ export class FactoryAPI {
     return new BigNumber(10 ** (18 - minDecimal.toNumber()));
   }
 
+  public async calculateRequiredComponentUnits(
```

Comment:
Won't this function return unrounded component amounts? Feel like it should only return whole numbers.